pilot/helpers/agents/Developer.py

```diff
@@ -87,36 +106,80 @@ def implement_task(self, i, development_task=None):
+        if self.project.dev_steps_to_load and 'breakdown.prompt' in self.project.dev_steps_to_load[0]['prompt_path']:
+            instructions = self.project.dev_steps_to_load[0]['llm_response']['text']
+            convo_dev_task.messages = self.project.dev_steps_to_load[0]['messages']
+            self.project.dev_steps_to_load.pop(0)
+        else:
+            instructions = convo_dev_task.send_message('development/task/breakdown.prompt', {
+                "name": self.project.args['name'],
+                "app_type": self.project.args['app_type'],
+                "app_summary": self.project.project_description,
+                "clarifications": self.project.clarifications,
+                "user_stories": self.project.user_stories,
+                "user_tasks": self.project.user_tasks,
+                "technologies": self.project.architecture,
+                "array_of_objects_to_string": array_of_objects_to_string,  # TODO check why is this here
+                "directory_tree": self.project.get_directory_tree(True),
+                "current_task_index": i,
+                "development_tasks": self.project.development_plan,
+                "files": self.project.get_all_coded_files(),
+                "task_type": 'feature' if self.project.finished else 'app'
+            })
 
         instructions_prefix = " ".join(instructions.split()[:5])
         instructions_postfix = " ".join(instructions.split()[-5:])
-        response = convo_dev_task.send_message('development/parse_task.prompt', {
-            'running_processes': running_processes,
-            'os': platform.system(),
-            'instructions_prefix': instructions_prefix,
-            'instructions_postfix': instructions_postfix,
-        }, IMPLEMENT_TASK)
+        if self.project.dev_steps_to_load and 'parse_task.prompt' in self.project.dev_steps_to_load[0]['prompt_path']:
+            response = json.loads(self.project.dev_steps_to_load[0]['llm_response']['text'])
+            convo_dev_task.messages = self.project.dev_steps_to_load[0]['messages']
+            remove_last_x_messages = 1  # reason why 1 here is because in db we don't store llm_response in 'messages'
+            self.project.dev_steps_to_load.pop(0)
+        else:
+            response = convo_dev_task.send_message('development/parse_task.prompt', {
+                'running_processes': running_processes,
+                'os': platform.system(),
+                'instructions_prefix': instructions_prefix,
+                'instructions_postfix': instructions_postfix,
+            }, IMPLEMENT_TASK)
+            remove_last_x_messages = 2
+
         steps = response['tasks']
-        convo_dev_task.remove_last_x_messages(2)
+        convo_dev_task.remove_last_x_messages(remove_last_x_messages)
 
         completed_steps = []
 
         while True:
+            # This whole if statement is loading of project.
```

Comment:
Should this be before (outside) of the while loop? As I understand it, it only needs to execute once (if loading)